src/gp_creds.c

```diff
@@ -563,7 +567,14 @@ static uint32_t gp_check_cred(uint32_t *min,
     if (lifetime == 0) {
         ret_maj = GSS_S_CREDENTIALS_EXPIRED;
     } else {
-        ret_maj = GSS_S_COMPLETE;
+        if (svc->min_lifetime && lifetime < svc->min_lifetime) {
+            GPDEBUG("%s: lifetime (%u) less than min_lifetime (%u) "
+                    "for service \"%s\" - returning GSS_S_CREDENTIALS_EXPIRED\n",
```

Comment:
Please drop the " - returning GSS_S_CREDENTIALS_EXPIRED" part, as that will be logged already as the return error of the function in debug mode. Replace with just "returning".

Reply:
Okay.